utlities.js

```diff
@@ -0,0 +1,18 @@
+/**
+ * this function adds and ingredient to the search
+ * to be more specific about the cake search
+ * it is connected to the radio buttons
+ *
+ * @param {object} e - submit event
+ */
+function searchIngredient(e) {
+  e.preventDefault();
+  inputElements.forEach((elem) => {
+    //empties the list of contents
+    ingredientsListElement = "";
+    if (elem.checked == true) {
```

Comment:
You don't need to loop through all the radios to get the option selected. Using `querySelector` one could get the checked  radio like this:
```javascript
const checkedRadio = document.querySelector( ("input[name='ingredient']:checked"); 
```
For more details see [here](https://www.geeksforgeeks.org/how-to-check-whether-a-radio-button-is-selected-with-javascript/#:~:text=Using%20Input%20Radio%20checked%20property,id%20is%20check%20or%20not.).

Reply:
cool